### PR TITLE
Rename cirq-dev to cirq-unstable

### PR DESCRIPTION
**Pull request by Strilanc**
- Refactor verify-published-package to have an --unstable flag.
- Refactor verify-published-package to require --prod/--test/--unstable be specified
- Rename various CIRQ_DEV type things to CIRQ_UNSTABLE type things

Fixes https://github.com/quantumlib/Cirq/issues/2431

### CODE REVIEW

**Comment by Strilanc** [comment]
At least until next time.

File: dev_tools/packaging/publish-dev-package.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
```
Comment (dabacon) [inline dev_tools/packaging/publish-dev-package.sh]: rename script since it now pushes to unstable?

**Comment by dabacon** [comment]
> this is the last time we change the name of this!

Said by every software engineer every time.

**Comment by balopat** [comment]
> > this is the last time we change the name of this!
> 
> Said by every software engineer every time.

Maybe https://github.com/quantumlib/Cirq/pull/3527 is the last time we rename this...?

**Pull request merged**